### PR TITLE
fix the problem that --tex_template flag didn't handle.

### PR DESCRIPTION
**Pull request by yalikes**
## Motivation

1. there is a --tex_template flag that allow us to specify tex_template file to be used.
but previously this flag takes no effect, because method digest_args() in ManimConfig class
didn't handle this flag. and I add related code.
2. in manim document,it says TexTemplateFromFile accpet a parameter 'tex_filename' to specify Path to a tex template file, but in code, it actually use 'filename'. We should follow document and name convention by change it to use 'tex_filename'.
## Oneline Summary of Changes

```
- Added new feature ... / Fixed bug ... / ... (:pr:`PR NUMBER HERE`)
```

## Acknowledgements
- [ ] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)

## Reviewer Checklist
- [ ] Newly added functions/classes are either private or have a docstring
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [ ] The oneline summary has been included [in the wiki](https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release)

### CODE REVIEW

File: manim/_config/utils.py [hunk 1]
```diff
@@ -692,6 +693,17 @@ def digest_args(self, args: argparse.Namespace) -> "ManimConfig":
...
+        if args.tex_template:
+            if args.tex_template == "ctex":
+                self.tex_template = TexTemplateLibrary.ctex
+            elif args.tex_template == "simple":
+                self.tex_template = TexTemplateLibrary.simple
+            elif args.tex_template == "threeb1b":
+                self.tex_template = TexTemplateLibrary.threeb1b
+            elif args.tex_template == "default":
+                self.tex_template = TexTemplateLibrary.default
+
```
Comment (leotrs) [inline manim/_config/utils.py:706]: @ManimCommunity/latex could you please check if this is the preferred way of handling this flag?

**Review by leotrs — COMMENTED** [review]
Woops! This is totally my fault - thanks for reporting and for submitting the fix!

File: manim/_config/utils.py (hunk 1 above)
Comment (PhilippImhof) [reply manim/_config/utils.py:706]: When first introduced, its aim was for the user to specify a custom template **file**, but I think with @cobordism's improvements, this is now the way to go. He knows better.

**Comment by cobordism** [comment]
The --tex_template CLI flag is there for backwards compatibility and is meant to specify a tex template file. 
It allows you to set your own custom global tex template file to be used by all tex mobjects.
Example: https://github.com/ManimCommunity/manim/blob/master/example_scenes/customtex.py


The preferred way for dealing with tex templates going forward is not as a CLI flag, but as a keyword argument to the tex mobjects in question like here:
https://docs.manim.community/en/v0.2.0/reference/manim.mobject.svg.tex_mobject.html#extra-latex-packages
or here: 
https://docs.manim.community/en/v0.2.0/reference/manim.mobject.svg.tex_mobject.html#latex-maths-fonts-the-template-library

Note that it is also possible to read a template from file within your manim scene by using TexTemplateFromFile: https://github.com/ManimCommunity/manim/blob/master/manim/utils/tex.py#L230

**Comment by yalikes** [comment]
> The --tex_template CLI flag is there for backwards compatibility and is meant to specify a tex template file.
> It allows you to set your own custom global tex template file to be used by all tex mobjects.

thanks for your explain!  yes, --tex_template flag should only uesd to specify tex template file.

**Review by cobordism — APPROVED** [review]

**Review by PhilippImhof — APPROVED** [review]
LGTM. Thank you

**Pull request merged**